de-lib/src/main/java/org/iplantc/de/diskResource/client/views/details/DetailsViewImpl.java

```diff
@@ -178,12 +179,14 @@ public HandlerRegistration addSubmitDiskResourceQueryEventHandler(SubmitDiskReso
     //<editor-fold desc="Event Handlers">
     @Override
     public void onDiskResourceSelectionChanged(DiskResourceSelectionChangedEvent event) {
+        mask(appearance.loadingMask());
         if (event.getSelection().isEmpty()
                 || event.getSelection().size() != 1 || event.getSelection().get(0).isFilter()) {
             bind(null);
             // Hide table
             table.addClassName(appearance.css().hidden());
             emptyDetails.removeClassName(appearance.css().hidden());
+            unmask();
             return;
         }
```

Comment:
Could the `mask` be moved after this `if` check? Then it wouldn't need the `unmask` inside?